ethcore/evm/benches/basic.rs

```diff
@@ -207,3 +230,188 @@ fn result(r: Result<evm::GasLeft>) -> U256 {
 		_ => U256::zero(),
 	}
 }
+
+/// Runs a given EVM bytecode.
+fn run_code(b: &mut Bencher, code: Bytes) {
+	let factory = Factory::default();
+	let mut ext = FakeExt::new();
+	b.iter(|| {
+		let mut params = ActionParams::default();
+		params.address = Address::from_str("0f572e5295c57f15886f9b263e2f6d2d6c7b5ec6").unwrap();
+		params.gas = U256::MAX;
+		params.code = Some(Arc::new(black_box(code.clone())));
+		let vm = factory.create(params, ext.schedule(), 0);
+		result(vm.exec(&mut ext).ok().unwrap())
+	});
+}
+
+/// Compute mulmod(U256::MAX, U256::MAX, 1) 500 times.
+fn mulmod1_500(b: &mut Criterion) {
+	b.bench_function("mulmod modulo 1, 500 times", |b| {
+		run_code(b, "6101f45b6001900360017fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff80095080600357".from_hex().unwrap());
+	});
+}
+
+/// Compute mulmod(U256::MAX, U256::MAX, 1) 1000 times.
```

Comment:
What's the point of having a loop in the bytecode with different number of iterations, just to make sure it scales linearly? IMHO we should be benchmarking a single operation and criterion can take care of running it in a loop. It would make sense to have a loop if we were using a JIT compiler, but we're not.

Reply:
Without a loop in the smart contract you get a significant noise from loading the contract. You are right, different numbers of iterations are to check that scaling is linear.